Receive Only Channels + Different Zone/Channel Announce Text vs. Display Text

crazyghost-12 opened the pull request:
**_Receive Only Channels_**
- This allows channels in a codeplug to be configured as receive only which prevents radios using that codeplug from transmitting on that channel.

**_ Different Zone/Channel Announce Text vs. Display Text_**
- This allows zone and channel names to announce text that is different than what it displays. (Ex: Radio could display channel name "LawDisp" but announce "Law Dispatch" when switching to that channel)

crazyghost-12 committed:
Add check for receive-only channels before transmission

crazyghost-12 committed:
Update voice announcements to use name_announce

crazyghost-12 committed:
Add name_announce value for zones and channels; add isReceiveOnly value for channels (prevents radios using this codeplug to transmit on that channel)

crazyghost-12 committed:
Rename receiveOnly property to isReceiveOnly

crazyghost-12 committed:
Forgot to update voice announce for changeChannel

firealarmss commented:
@Firav looks good from what I can tell. Thoughts?

Firav commented:
@firealarmss Did we ever fix the API issue we had with responsivevoice.org? We got ratelimited or something, thats why we had to disable it in the first place.

Firav reviewed (changes requested):
If we're going to re-enable responsiveVoice, we'll need to enable these lines again as well. Looks like you missed these for the scanner models too:
656-657
if (!isScannerModel()) {
            // responsiveVoice.speak(`${currentZone.name}`, `US English Female`, {rate: .8});
            // responsiveVoice.speak(`${currentChannel.name}`, `US English Female`, {rate: .8});
}

crazyghost-12 committed:
Forgot to update zone/channel announce for scannerMode

crazyghost-12 commented:
@Firav,
Updated, sorry!

Firav reviewed (changes requested):
https://github.com/WhackerLink/WhackerLinkFiveM/blob/bd458282115b6b7eae7c27ce7f8039d3b3882dca/client/ui/index.html#L110
This line will need to be uncommented as well, else responsiveVoice doesnt work at all.

Firav commented:
> @firealarmss Did we ever fix the API issue we had with responsivevoice.org? We got ratelimited or something, thats why we had to disable it in the first place.

@firealarmss Do we want to just make the responsivevoice API key a user-configurable? Like we did with AVD?
Basically a bring-your-own-license model.

crazyghost-12 committed:
Enable responsive voice script in index.html

firealarmss commented:
Yeah I think we should make this configurable so we don't hit a rate limit again. @Firav

crazyghost-12 committed:
Add new configuration options to config.yml

crazyghost-12 committed:
Remove announceZoneChannelChanges from config

Remove the announceZoneChannelChanges configuration option.

crazyghost-12 committed:
Add announceZoneChannelChanges to codeplug example

Added 'announceZoneChannelChanges' field to configuration.

crazyghost-12 committed:
Remove comment on announceZoneChannelChanges

Removed comment about enabling announceZoneChannelChanges.

crazyghost-12 committed:
Remove responsiveVoiceApiKey from config.yml

Removed responsiveVoiceApiKey placeholder from config.

Firav commented:
> Yeah I think we should make this configurable so we don't hit a rate limit again. @Firav

@crazyghost-12 would that be some you'd like to implement, or would you like me to create another PR for it?

crazyghost-12 commented:
> > Yeah I think we should make this configurable so we don't hit a rate limit again. @Firav
> 
> @crazyghost-12 would that be some you'd like to implement, or would you like me to create another PR for it?

I will attempt it!

crazyghost-12 committed:
Add responsiveVoiceApiKey to config

crazyghost-12 committed:
Add isAnnonceZoneChannelChanges field to example config

crazyghost-12 committed:
Add API key check for voice announcements

Fetch and check responsiveVoiceApiKey from config.yml before announcing zone and channel.

crazyghost-12 commented:
@Firav Can you look over lines 661-682? Before I do the rest..

Firav commented:
> @Firav Can you look over lines 661-682? Before I do the rest..

Yikes. That'll work in theory yeah, but we dont want to do that for every entry that uses responsiveVoice. Try making a function somewhere in the script.js that retrieves that information and stores it as a global variable that you can use later.

crazyghost-12 commented:
> > @Firav Can you look over lines 661-682? Before I do the rest..
> 
> 
> 
> Yikes. That'll work in theory yeah, but we dont want to do that for every entry that uses responsiveVoice. Try making a function somewhere in the script.js that retrieves that information and stores it as a global variable that you can use later.

Ok sounds good, I'll try that. Sorry I have very very little coding experience and really just know the basics so I'm just kinda doing this to challenge myself. I'll give that a shot when I get off work tonight!

crazyghost-12 committed:
Implement responsive voice API key handling

Added logic to fetch and check responsiveVoiceApiKey from config.yml, enabling/disabling responsive voice based on its presence.

crazyghost-12 commented:
@Firav

How do lines 133-152 look (for the global var) and then 683-686 for checking the global var to announce the channel/zone? If everything looks good, I'll implement it to the other places where responsivevoice.speak is triggered and update the HTML file to include the config file's API key in the script header for responsive voice.

firealarmss commented:
If I recall correctly, I pass the entire parsed config option into the script.js on start. Is there a reason for reopening and parsing the config?

crazyghost-12 committed:
Refactor config loading and API key handling

crazyghost-12 commented:
> If I recall correctly, I pass the entire parsed config option into the script.js on start. Is there a reason for reopening and parsing the config?

@Firav How about now... lines 104-141?

Firav commented:
> > If I recall correctly, I pass the entire parsed config option into the script.js on start. Is there a reason for reopening and parsing the config?
> 
> @Firav How about now... lines 104-141?

Looking good! Just gotta finish out your logic for the rest of the responsiveVoice API calls and for the actual API key itself in the script inclusion. Also marked a change for you to review.

crazyghost-12 committed:
Add conditional checks for voice announcements

crazyghost-12 commented:
@Firav The remaining placed where responsiveVoice.speak were updated the only thing is, I'm not sure how to pull a YAML file value into HTML, so I was wondering if you'd be able to help me or do it yourself? Also, you mentioned you marked a change for me... where do I see that?

firealarmss reviewed (commented):
I don’t think I like that YAML parsing? @Firav ?
